src/parser.cpp

```diff
@@ -1125,4 +1157,216 @@ void Parser::ParserImpl::loadImport(const ImportSourcePtr &importSource, const M
     }
 }
 
+void Parser::ParserImpl::loadReset(const ResetPtr &reset, const ComponentPtr &component, const XmlNodePtr &node)
+{
+    int order = 0;
+    bool orderDefined = false;
+    VariablePtr referencedVariable = nullptr;
+    std::string variableName = "";
+
+    XmlAttributePtr attribute = node->getFirstAttribute();
+    while (attribute) {
+        if (attribute->isType("variable")) {
+            const std::string variableReference = attribute->getValue();
+            referencedVariable = component->getVariable(variableReference);
+            if (referencedVariable == nullptr) {
+                ErrorPtr err = std::make_shared<Error>();
+                err->setDescription("Reset referencing variable '" + variableReference +
+                                    "' is not a valid reference for a variable in component '" + component->getName() + "'.");
+                err->setReset(reset);
+                err->setKind(Error::Kind::RESET);
+                err->setRule(SpecificationRule::RESET_INVALID_VARIABLE_REFERENCE);
+                mParser->addError(err);
+            } else {
+                reset->setVariable(referencedVariable);
+            }
+        } else if (attribute->isType("order")) {
+            if (convertToInt(attribute->getValue(), &order)) {
+                orderDefined = true;
+            } else {
+                if (reset->getVariable() != nullptr) {
+                    variableName = reset->getVariable()->getName();
+                }
+                ErrorPtr err = std::make_shared<Error>();
+                err->setDescription("Reset in component '" + component->getName() +
+                                    "' referencing variable '" + variableName +
```

Comment:
We ought to handle the case where `variableName` is a null string.

Reply:
Null string or empty string, I'm not clear on what you mean here.

Reply:
By null string, I indeed mean an empty string. `variableName` can be equal to `""`, which means that the error message would be a bit nonsensical.